File: 2460_apply_operations_to_an_array.java
class Solution { 
    public int[] applyOperations(int[] x) { 
        int y = x.length, z = 0; 
        for(int i = 0; i < y - 1; i++) { 
            if(x[i] != 0 && x[i] == x[i + 1]) { 
                x[i] *= 2; 
                x[i + 1] = 0; 
            } 
        } 
        int[] w = new int[y]; 
        for(int i = 0; i < y; i++) if(x[i] != 0) w[z++] = x[i]; 
        return w; 
    } 
}

File: 2460_apply_operations_to_an_array.cpp
class Solution { 
public: 
    vector<int> applyOperations(vector<int>& x) { 
        int y = x.size(), z = 0; 
        for(int i = 0; i < y - 1; i++) 
            if(x[i] && x[i] == x[i + 1]) x[i] *= 2, x[i + 1] = 0; 
        vector<int> w(y); 
        for(int i = 0; i < y; i++) if(x[i]) w[z++] = x[i]; 
        return w; 
    } 
};
